text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,877 @@
+const oac = new cloudfront.OriginAccessControl(this, 'myS3OAC');
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: {
+    origin: new origins.S3Origin(myBucket, {
+      originAccessControl: oac
+    })
+  },
+});
+```
+
+It is recommended to set the `@aws-cdk/aws-cloudfront:useOriginAccessControlByDefault` feature flag to `true`, so an OAC will be automatically created instead
+of an OAI when `S3Origin` is instantiated. If you don't set this feature flag, and OAI will be created and granted access to the underlying bucket.
+
+## Migrating from OAI to OAC
+
+If you are currently using OAI for your S3 origin and wish to migrate to OAC, first set the feature flag `@aws-cdk/aws-cloudfront:useOriginAccessControlByDefault`
+to `true` in `cdk.json`. With this feature flag set, when you create a new `S3Origin` an Origin Access Control will be used instead of Origin Access Identity.
+You can create and pass in an `OriginAccessControl` or one will be automatically created by default. Run `cdk diff` before deploying to verify the
+changes to your stack.
+
+For more information, see [Migrating from origin access identity (OAI) to origin access control (OAC)](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-restricting-access-to-s3.html#migrate-from-oai-to-oac).
+
+### Using pre-existing S3 buckets
+
+If you are using an imported bucket for your S3 Origin and want to use OAC, first import the bucket using one of the import methods (`fromBucketName`,
+`fromBucketArn` or `fromBucketAttributes`).
+
+To update the bucket policy to allow CloudFront access you can set the `overrideImportedBucketPolicy` property to `true`. The `S3Origin` construct
+will update the S3 bucket policy by appending the following policy statement to allow CloudFront read-only access:
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Sid": "GrantOACAccessToS3",
+        "Effect": "Allow",
+        "Principal": {
+            "Service": "cloudfront.amazonaws.com"
+        },
+        "Action": "s3:GetObject",
+        "Resource": "arn:aws:s3:::<S3 bucket name>/*",
+        "Condition": {
+            "StringEquals": {
+                "AWS:SourceArn": "arn:aws:cloudfront::111122223333:distribution/<CloudFront distribution ID>"
+            }
+        }
+    }
+}
+```
+
+> Note: If your bucket previously used OAI, you will need to manually remove the policy statement
+that gives the OAI access to your bucket from your bucket policy.
+
+```ts
+const bucket = s3.Bucket.fromBucketArn(this, 'MyExistingBucket', 
+  'arn:aws:s3:::mybucketname'
+);
+
+const oac = new cloudfront.OriginAccessControl(this, 'MyOAC', {
+  originAccessControlOriginType: cloudfront.OriginAccessControlOriginType.S3,
+});
+
+const distribution = new cloudfront.Distribution(this, 'MyDistribution', {
+  defaultBehavior: {
+    origin: new origins.S3Origin(bucket, {
+      originAccessControl: oac,
+      overrideImportedBucketPolicy: true
```

Comment:
Like @TheRealAmazonKendra mentioned above, the prop `overrideImportedBucketPolicy` cannot function in the current CDK. This is because all `from*` methods don't perform a CFN resource import, instead they only exist so CDK users can refer to their imported bucket in a type-safe way. They aren't part of the stack, they appear as raw ARN strings in the template. This means that CFN, and therefore CDK, has no way to modify these imported resources.